### ISSUE TITLE
Cross-dimention aggro and player killing

### ISSUE THREAD

**Issue by Malhera** [PaperMC/Paper#3981]
### What behaviour is expected:
When angering zombified piglins, you expect them to be angry at you, and follow you around the nether.

### What behaviour is observed:
**You don't expect them to also follow your overworld-movements FROM THE NETHER** coordinate-wise to a 1:1 ratio, nor do you expect them, if they find a place they can stand in melee range to you, coordinate-wise, you in the overworld and them in the nether, to actually be able to HIT you in melee range, and kill you cross dimensions!

The melee animation doesn't play, and bed-respawn doesn't work, you respawn at the map's origin as if you died in the nether.

We "chased" this bug on my server, trying to understand how it works. The only temporary fix we found was to bedrock-fill areas of the nether that Zpiglins can reach that coincide "too much" with coordinates of the overworld that we heavily use.

### Steps/models to reproduce:
How to observe it, with two players/accounts.
-player1 stays in the nether, possibly in creative/spectator as to observe without interference
-player2 is in survival/adventure mode.
-player2 make zombified piglins antsy, and they aren't all dead
-player2 go through nether portal
-player2 moves around in the overworld
-player1 should be able to see Zpiglins angry-follow "an invisible something", this something being the 1:1 position of the player2 in the overworld, but in the nether
-if player 2 is at lets say x50 y64 z50, and angry Zpiglin can reach/approach in melee range these coordinates in the nether, they would be able to hit and kill player2 cross dimension.
-the message "Player2 has been slain by Zombified Piglin" should be displayed.
-player1 should observe the piglins loose aggro/interest/ disperse after player2's death.

### Plugin list:

None of the plugins should modify mob/AI behavior (but Shopkeepers), and none have anything to do with dimension/world thread management.

AdvancedHelp
ArmorStandEditor
AztecTabCompleter
Billboard - disabled
ChatMessenger
CoreProtect
CraftBook
dynmap
Lagmonitor
LibsDisguises
LuckPerms
MapManager (has to do with MC maps, not MCworlds, don't loose time on that)
MarriageMaster
openInv
PacketListenerApi
PlayerHeads
ProtocolLib
ServerlistMOTD
Shopkeepers
TheNewEconomy - disabled, not updated yet
Vault
Voteban
WhitelistPlus
WorldEdit

### Paper build number:
1.16.1:
Observed first with Paper-79
Actually running Paper-101, still occuring, same exact behavior.

### Anything else:
Mmmh, I know I'm not using the lastest java version (from a long shot). FYI the xubuntu VM in which the servers runs still have java 1.8.0_222 openjdk 64bits.
I don't know if it has an influence on this behavior issue, but it doesn't hurt to mention it.

**Comment by Malhera** [comment]
Post Scriptum: I suspect it would be linked/twined to the https://github.com/PaperMC/Paper/issues/3879 issue report, which seems to indicate player-entity information leaking/duplicating cross-dimensions.


I would gladly have tried running multiverse to kinda split the world threads to see if it could be a work around, but MV is still in 1.15, and I don't feel too much like adding it to my server, we don't have use of it.

**Comment by Proximyst** [comment]
Can reproduce without plugins. Funny bug, good catch!

**Comment by Malhera** [comment]
Thanks!

But funny? Not really, actually caused quite a bit of disorder around, as players were killed by untraceable enemies, that no spectator mode nor use of adding glowing effect to nearby entities, nor plugin-detection, would succeed at making visible/appear.

So much for the feeling of being haunted or cursed!

Even more as the server is in hardcore mode!! These zombified piglins hit hard!

I just hope it get fixed so we can regen the areas of the nether we had to bed-rock fill.

**Comment by Phoenix616** [comment]
> I would gladly have tried running multiverse to kinda split the world threads to see if it could be a work around, but MV is still in 1.15, and I don't feel too much like adding it to my server, we don't have use of it.

Just fyi: Multiverse works fine in 1.16 but I highly doubt it will influence this issue as something like "world threads" does not exist.

**Comment by Malhera** [comment]
...Which was just a illustrative way to say that *maybe* having a plugin handle world loadings/unloadings, and mv-portal manage teleportations, that the tp event to an other dimension would make the mob's aggro drop properly somehow.

I don't know.
I'm not a paper/plugin dev, when it comes to java the most I did was dirty-update a deprecated plugin I liked for myself, and it wasn't a really complex one, and it barelly works, but just enough for my use. I just did 'basic' stuff when it comes to Java, and I honestly don't actually want to dive into it.

And thanks for the heads up about Multiverse working in 1.16, but point is, we don't really need it anyway, and since according to you it wouldn't influence the issue, there isn't really a point in getting it on this server.
